src/main/scala/controllers/ActionHandler.scala

```diff
@@ -103,13 +103,12 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
                         sectionNumber: Int,
                         section: Option[ApplicationSection],
                         questions: Map[String, Question],
-                        fields: Seq[Field],
                         errs: FieldErrors,
                         hints: FieldHints): Future[Result] = {
     val answers = section.map { s => s.answers }.getOrElse(JsObject(Seq()))
 
     gatherSectionDetails(id, sectionNumber).map {
-      case Some((app, appForm, formSection, opp)) => selectSectionForm(sectionNumber, section, formSection.questionMap, answers, fields, errs, app, appForm, opp)
+      case Some((app, appForm, formSection, opp)) => selectSectionForm(sectionNumber, section, formSection.questionMap, answers, formSection.fields, errs, app, appForm, opp)
```

Comment:
I'll break that line up - it's definitely getting too long, but it highlights the fact that it's taking 9 parameters, which is a smell in itself.